pkg/api/validation/validation.go

```diff
@@ -647,6 +647,23 @@ func ValidatePodUpdate(newPod, oldPod *api.Pod) errs.ValidationErrorList {
 	return allErrs
 }
 
+// ValidatePodStatusUpdate tests to see if the update is legal for an end user to make. newPod is updated with fields
+// that cannot be changed.
+func ValidatePodStatusUpdate(newPod, oldPod *api.Pod) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+
+	allErrs = append(allErrs, ValidateObjectMetaUpdate(&oldPod.ObjectMeta, &newPod.ObjectMeta).Prefix("metadata")...)
+
+	// TODO: allow change when bindings are properly decoupled from pods
```

Comment:
If we ever wanted to allow scheduler to mutate status instead of binding, but no other reason.  I'll remove the todo


> On Feb 26, 2015, at 11:09 PM, Brian Grant <notifications@github.com> wrote:
> 
> In pkg/api/validation/validation.go:
> 
> > @@ -647,6 +647,23 @@ func ValidatePodUpdate(newPod, oldPod *api.Pod) errs.ValidationErrorList {
> >  	return allErrs
> >  }
> >  
> > +// ValidatePodStatusUpdate tests to see if the update is legal for an end user to make. newPod is updated with fields
> > +// that cannot be changed.
> > +func ValidatePodStatusUpdate(newPod, oldPod *api.Pod) errs.ValidationErrorList {
> > +	allErrs := errs.ValidationErrorList{}
> > +
> > +	allErrs = append(allErrs, ValidateObjectMetaUpdate(&oldPod.ObjectMeta, &newPod.ObjectMeta).Prefix("metadata")...)
> > +
> > +	// TODO: allow change when bindings are properly decoupled from pods
> In what situation would we allow changes? Pods don't get rescheduled, and I'm thinking that the cleanest way to deal with rejection from Kubelet is to consider the pod dead.
> 
> —
> Reply to this email directly or view it on GitHub.
>